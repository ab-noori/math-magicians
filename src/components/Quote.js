import React, { useState, useEffect } from 'react';

const Quote = () => {
  const api = 'https://api.api-ninjas.com/v1/quotes?category=design';
  const key = 'GIDZtRlO9nOuhs77+bdOLg==6DdOB8VgjtrE29Vt';
  const [data, setData] = useState({});
  const [isLoading, setIsLoading] = useState(true);
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const response = await fetch(api, {
          headers: {
            'X-Api-Key': key,
          },
        });
        const data = await response.json();
        setData(data[0]);
        setIsLoading(false);
      } catch (error) {
        console.error(error);
        setIsLoading(false);
        setHasError(true);
      }
    };
    fetchData();
  }, []);

  if (isLoading) {
    return <div>Loading...</div>;
  }

  if (hasError) {
    return <div>Error while fetching data</div>;
  }

  return (
    <div className="quote">
      <h1>{data.category}</h1>
      <p>{data.quote}</p>
      <small>{data.author}</small>
    </div>
  );
};

export default Quote;
