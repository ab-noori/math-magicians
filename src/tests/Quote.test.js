import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

describe('Quote component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
