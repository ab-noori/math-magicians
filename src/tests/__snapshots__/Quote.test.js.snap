// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders correctly 1`] = `
<div>
  Loading...
</div>
`;
